sql/src/main/java/io/crate/operation/reference/sys/node/NodeOsCgroupStatsExpression.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.reference.sys.node;
+
+import org.elasticsearch.monitor.os.OsStats;
+
+import java.util.Map;
+
+public class NodeOsCgroupStatsExpression extends NestedNodeStatsExpression {
+
+    private static final String CPUACCT = "cpuacct";
+    private static final String CPU = "cpu";
+
+    public NodeOsCgroupStatsExpression() {
+        childImplementations.put(CPUACCT, new NodeOsCgroupCpuAcctStatsExpression());
+        childImplementations.put(CPU, new NodeOsCgroupCpuStatsExpression());
+    }
+
+    @Override
+    public Map<String, Object> value() {
+        if (row.osStats().getCgroup() != null) {
+            return super.value();
+        }
+        return null;
+    }
+
+    private class NodeOsCgroupCpuAcctStatsExpression extends NestedNodeStatsExpression {
+
+        private static final String CONTROL_GROUP = "control_group";
+        private static final String USAGE_NANOS = "usage_nanos";
+
+        public NodeOsCgroupCpuAcctStatsExpression() {
+            childImplementations.put(CONTROL_GROUP, new SimpleNodeStatsExpression<String>() {
```

Comment:
Would it make sense to introduce a different base class that takes care if the cgroup handling?
Something like

`CgroupExpression.forAttribute(Cgroup::getCpuAcctControlGroup)`

```
CgroupExpression extends RowContextCollectorExpression<NodeStatsContext, R> {


    public CgroupExpression(Function<OsStats.Cgroup, R> getter) {
        ...
    }

    @Override
    public R value() {
        if (row.isComplete()) {
            OsStats.Cgroup cgroup = row.osStats().getCgroup();
            if (cgroup == null) {
                return null;
            }
            return getter.apply(cgroup);
        }
        return null;
    }
```